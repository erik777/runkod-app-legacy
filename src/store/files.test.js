import reducer, {
  fetchAct,
  fetchedAct,
  fetchErrorAct
} from './files';

import {
  logOutAct
} from './user';

import {
  selectAct as projectSelectAct
} from "./project";

let state = undefined;

it('1- default', () => {
  expect(reducer(state, {})).toMatchSnapshot();
});

it('2- fetch', () => {
  const act = fetchAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('3- fetch error', () => {
  const act = fetchErrorAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('4- fetched', () => {
  const files = [
    {attrs: {parent: '/', label: 'index.html'}},
    {attrs: {parent: '/css/', label: 'style.css'}},
    {attrs: {parent: '/css/', label: 'vendor.css'}},
    {attrs: {parent: '/js/', label: 'main.js'}},
    {attrs: {parent: '/mobile/css/', label: 'mobile.css'}}
  ];
  const act = fetchedAct(files);
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('5- user logout', () => {
  const act = logOutAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});


it('6- project select', () => {
  const files = [
    {attrs: {parent: '/', label: 'index.html'}},
  ];

  state = reducer(state, fetchedAct(files));

  const act = projectSelectAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});
