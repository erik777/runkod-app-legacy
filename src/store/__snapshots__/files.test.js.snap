// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default 1`] = `
Object {
  "list": Array [],
  "loading": false,
  "map": null,
}
`;

exports[`2- fetch 1`] = `
Object {
  "list": Array [],
  "loading": true,
  "map": null,
}
`;

exports[`3- fetch error 1`] = `
Object {
  "list": Array [],
  "loading": false,
  "map": null,
}
`;

exports[`4- fetched 1`] = `
Object {
  "list": Array [
    Object {
      "label": "index.html",
      "parent": "/",
    },
    Object {
      "label": "style.css",
      "parent": "/css/",
    },
    Object {
      "label": "vendor.css",
      "parent": "/css/",
    },
    Object {
      "label": "main.js",
      "parent": "/js/",
    },
    Object {
      "label": "mobile.css",
      "parent": "/mobile/css/",
    },
  ],
  "loading": false,
  "map": Object {
    "/": Object {
      "files": Array [
        Object {
          "label": "index.html",
          "parent": "/",
        },
      ],
      "folders": Array [
        "/css/",
        "/js/",
        "/mobile/",
      ],
    },
    "/css/": Object {
      "files": Array [
        Object {
          "label": "style.css",
          "parent": "/css/",
        },
        Object {
          "label": "vendor.css",
          "parent": "/css/",
        },
      ],
      "folders": Array [],
    },
    "/js/": Object {
      "files": Array [
        Object {
          "label": "main.js",
          "parent": "/js/",
        },
      ],
      "folders": Array [],
    },
    "/mobile/": Object {
      "files": Array [],
      "folders": Array [
        "/mobile/css/",
      ],
    },
    "/mobile/css/": Object {
      "files": Array [
        Object {
          "label": "mobile.css",
          "parent": "/mobile/css/",
        },
      ],
      "folders": Array [],
    },
  },
}
`;

exports[`5- user logout 1`] = `
Object {
  "list": Array [],
  "loading": false,
  "map": null,
}
`;

exports[`6- project select 1`] = `
Object {
  "list": Array [],
  "loading": false,
  "map": null,
}
`;
